app/src/main/java/com/github/multimatum_team/multimatum/activity/DeadlineDetailsActivity.kt

```diff
@@ -82,6 +101,14 @@ class DeadlineDetailsActivity : AppCompatActivity() {
 
     }
 
+    private fun retrieveNotificationsTimes(): ArrayList<Long> {
+        val res = ArrayList<Long>()
+        for (checkBox in checkBoxIdTime) {
+            if (findViewById<CheckBox>(checkBox.key).isChecked) res.add(checkBox.value)
+        }
+        return res
+    }
```

Comment:
```suggestion
    private fun retrieveNotificationsTimes(): List<Long> = checkBoxIdTime.filter { checkBox -> findViewById<CheckBox>(checkBox.key).isChecked }
```

Reply:
Yeah, but I think you're not getting the values only here. There must be for sure a way to do it in functionnal prog, but I prefer to leave it like that. I find it more clear.